ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -581,6 +581,37 @@ public void testRangeRequestCompositeBlob()
     testRangeRequestFromStartOffset(maxChunkSize + maxChunkSize / 2, true);
   }
 
+  @Test
+  public void testEarlyReadableStreamChannelClose()
+      throws Exception {
+    final AtomicReference<Exception> callbackException = new AtomicReference<>();
+    Callback<GetBlobResult> callback = new Callback<GetBlobResult>() {
+      @Override
+      public void onCompletion(GetBlobResult result, Exception exception) {
+        if (exception != null) {
+          callbackException.set(exception);
+        } else {
+          try {
+            result.getBlobDataChannel().close();
+          } catch (IOException e) {
+            callbackException.set(e);
+          }
+        }
+      }
+    };
+
+    doPut();
+    GetBlobOperation op = createOperationAndComplete(callback);
+
+    if (callbackException.get() != null) {
+      throw callbackException.get();
+    }
+    Exception operationException = op.getOperationException();
```

Comment:
This test closes the `ReadableStreamChannel` when the router callback is called.  It then checks that the operation was aborted with a `ChannelClosed` router error.